include/aspect/material_model/interface.h

```diff
@@ -779,6 +780,25 @@ namespace aspect
 
 
 
+    /**
+     * For multicomponent material models: Given a vector of of compositional
+     * fields of length N, this function returns a vector of volume fractions
+     * of length N+1, corresponding to the volume fraction of a ``background
+     * material'' as the first entry, and volume fractions for each of the input
+     * fields as the following entries. The returned vector will sum to one.
+     * If the sum of the compositional_fields is greater than
+     * one, we assume that there is no background mantle (i.e., that field value
+     * is zero). Otherwise, the difference between the sum of the compositional
+     * fields and 1.0 is assumed to be the amount of background mantle.
+     * Optionally, one can input a component mask that determines, which of the
+     * compositional fields to use during the computation (e.g. because
+     * some fields do contain non-volumetric quantities like strain,
```

Comment:
I would drop 'do' here because it puts emphasis on something that I find misleading.